client/src/leap/soledad/client/api.py

```diff
@@ -70,6 +71,26 @@
 """
 SOLEDAD_CERT = None
 
+# A whitelist of modules from where to collect plugins dynamically.
+# For the moment restricted to leap namespace, but the idea is that we can pass
+# other "trusted" modules as options to the initialization of soledad.
```

Comment:
uh, I see now that I hadn't properly refreshed the PR page so I had missed some comments.

> That function would be called by leap.mail and leap.keymanager explicitelly to register their intention to plug modules in soledad, instead of the opposite. How does that sound?

so, there are two different things here:

https://twistedmatrix.com/documents/current/core/howto/plugin.html#alternate-plugin-packages

since we're not implementing a twisted interface, it looks like we need to pass a list of modules.
we could traverse the whole path, but I think that's overkill.

on the other hand, I'm still not quite comfortable with the idea of any package installed being able to potentially access the sync info. I know it's somehow irrational, since if the user has managed to install a malicious package he's owned all the same, but I see more reasons beyond security.

for instance, I think that some plugins in an imaginary ecosystem could be deemed more stable than others. think for example of their ability to block the reactor. we could think of some "review" process to flag some plugins as stable. the user would be free to configure his installation to choose "experimental" or "stable" plugins... something like firefox or alike. 

that's why I opted for a whitelist-by-default to begin with, plus some mechanism that could allow some kind of policy to include plugins from certain packages (or, whole namespaces: leap.*, pixelated.* etc). note that the namespace doesn't mean it's controlled by us.

eventually, if this prospers, we could have some interface to enable/disable plugins, in the main client, or in a cli that would change config at runtime in the soledad daemon.

regarding over-engineering, well, I like the api, it's well tested since it's used heavily in twisted itself (vs any roll-our-own), and it's already in our dependency tree, so we're not including any new dep.